### PR TITLE
Support downloading list of files in Resteasy Reactive Multipart

### LINKED ISSUE #26164: Quarkus Reactive support download multiple files multipart/form-data
### Description

Quarkus Reactive supports multipart response thanks to the #21038 commit.

The download feature is incomplete, at least compared to the upload capability. In particular, I would like multiple files with their filename under the same `content-disposition` name to be generated as a list of `content-disposition` on the wire. I desire `List<FileDownload>` attribute in my multipart DTO to be supported. 

The code of `MultipartMessageBodyWriter` seems to cover only the multipart DTO with a simple static attribute (not a list), not a list of `FileDownload`. On the contrary, the upload supports lists of `FileUpload`. So the current implementation is not symmetric.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by Sgitario**
Additionally, it adds support of DownloadFile objects. 

Fix https://github.com/quarkusio/quarkus/issues/26164

### CODE REVIEW

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/multipart/PartItem.java [hunk 1]
```diff
@@ -24,4 +26,8 @@ public MediaType getType() {
     public Object getValue() {
         return value;
     }
+
+    public String getRawGenericType() {
```
Comment (geoand) [inline independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/multipart/PartItem.java:30]: I don't think this name properly conveys what this does.
At the very least, we should probably document what it returns for both generic and non generic cases.

**Review by geoand — COMMENTED** [review]
LGTM, just added one comment

File: independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/multipart/PartItem.java (hunk 1 above)
Comment (Sgitario) [reply independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/multipart/PartItem.java]: Renamed to "firstParamType" and added comments in the method.

Comment (Sgitario) [reply independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/multipart/PartItem.java]: Let me know if the name looks better for you

**Review by geoand — APPROVED** [review]
LGTM, thanks a lot!

**Comment by eye0fra** [comment]
Awesome! Thanks a lot! @Sgitario do you have a sample of the raw response of the ### multi-file download?

**Comment by Sgitario** [comment]
> Awesome! Thanks a lot! @Sgitario do you have a sample of the raw response of the ### multi-file download?

You can see the raw response by running this test: https://github.com/quarkusio/quarkus/pull/26309/files#diff-136f284323fc8e3022c3f73dce4ed1d4e1ce5d4adf1c6e9affe221f6b0c6befdR102

Either read the `response` String, or add `.log().all()` after the line 106.

**Review by mamadaliev — APPROVED** [review]

**Pull request merged**